core/services/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -345,19 +341,27 @@ func (eb *EthBroadcaster) handleInProgressEthTx(etx EthTx, attempt EthTxAttempt,
 	}
 	parentCtx := context.TODO()
 
-	if etx.Simulate {
-		simulationCtx, cancel := context.WithTimeout(parentCtx, SimulationTimeout)
-		defer cancel()
-		if b, err := simulateTransaction(simulationCtx, eb.ethClient, attempt, etx); err != nil {
-			if jErr := eth.ExtractRPCError(err); jErr != nil {
-				eb.logger.CriticalW("Transaction reverted during simulation", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "err", err, "rpcErr", jErr.String(), "returnValue", b.String())
-				etx.Error = null.StringFrom(fmt.Sprintf("transaction reverted during simulation: %s", jErr.String()))
-				return eb.saveFatallyErroredTransaction(&etx)
-			}
-			eb.logger.Warnw("Transaction simulation failed, will attempt to send anyway", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "err", err, "returnValue", b.String())
-		} else {
-			eb.logger.Debugw("Transaction simulation succeeded", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "returnValue", b.String())
-		}
+	meta, err := etx.GetMeta()
+	if err != nil {
+		return errors.Wrap(err, "parsing tx meta")
+	}
+
+	checker, err := buildChecker(meta, eb.ethClient)
+	if err != nil {
+		return errors.Wrap(err, "building transmit checker")
+	}
+
+	shouldTransmit, err := checker.ShouldTransmit(parentCtx, eb.logger, etx, attempt)
```

Comment:
We want to inject a timeout here in addition to the parent ctx

Reply:
Passing in a context that has a timeout now which I think should work for both simulate and VRF